graphql/schema.graphql

```diff
@@ -183,11 +183,21 @@ type BitbucketProject {
   id: Int
 }
 
+type BitbucketClone {
```

Comment:
Why not call this type `BitbucketLink` and add a `type` property to it which will be a value from an enum?

As of now, the `BitbucketLinks` type looks redundant, it's just a wrapper for an array

Reply:
> Why not call this type `BitbucketLink` and add a `type` property to it which will be a value from an enum?
> 
> As of now, the `BitbucketLinks` type looks redundant, it's just a wrapper for an array

@ron-rookout I disagree. I think emums do not logically represent the added structure and will be confusing. Representing the api's structure with graphql objects will be more readable imo.

Reply:
What's wrong with

```graphql
enum LinkType {
     CLONE
//Other types that might be in the future
}

type BitbucketLink {
  href: String
  name: String
  type: LinkType
}

```

And then have a `[BitbucketLink]` be returned?

Reply:
But this is not the return structure. This is how it returns:
![image](https://user-images.githubusercontent.com/95576653/183654183-c9210e3a-713f-41e6-ac93-748b5cff5554.png)

Reply:
Your proposal would fit if it would be structured like:
{
links: [
"name": "...",
"href": "...",
"type": "clone"
}

Reply:
We shouldn't care what returns from external apis, we need to fit it for our needs. Explorook isn't simply a proxy to the bitbucket on prem API, we can reshape the data to a form that will work better for us.

Reply:
> We shouldn't care what returns from external apis, we need to fit it for our needs. Explorook isn't simply a proxy to the bitbucket on prem API, we can reshape the data to a form that will work better for us.

But we currently don't process the data. Besides, the frontend already uses that structure when processing saas. Restructuring will require us to refactor both the frontend and explorook just to make the data structure a bit cleaner. This will make us use proprietary self-maintained structures. I think we should just use what we get as it is, it is not that bad.

Reply:
> We shouldn't care what returns from external apis, we need to fit it for our needs. Explorook isn't simply a proxy to the bitbucket on prem API, we can reshape the data to a form that will work better for us.

Another thing. Explorook prrtty much is a proxy for the bitbucket api. The only reason we even use it is due to cors limitations